### PR TITLE
Feature/6

### PR DESCRIPTION
**Pull request by daniel-rocha**
Added LoadPatient endpoint, controller unit tests, integration tests.

### CODE REVIEW

File: src/PatientTestsApi/Test/Services/PatientDataService.spec.ts [hunk 1]
```diff
@@ -25,6 +25,30 @@ describe("PatientDataService #integration", async function (): Promise<void> {
...
+    const id = await dataService.insertPatient(expectedPatient);
+
+    // test
+    const createdPatient = await dataService.findPatient(id);
+    Object.keys(expectedPatient).forEach(key => {
+      expect(createdPatient[key]).deep.equal(expectedPatient[key]);
+    });
+    expect(createdPatient._id).is.equal(id);
+    expect(createdPatient._shardKey).is.equal(id);
+  });
```
Comment (hannesne) [inline src/PatientTestsApi/Test/Services/PatientDataService.spec.ts:50]: findPatient should not be returning database properties. (_id and _shardKey)
We can merge the find and create tests into one test called "Can find and create a patient.". Pull the following 
```
const createdPatient = await db.loadPatient(id);
    Object.keys(expectedPatient).forEach(key => {
      expect(createdPatient[key]).deep.equal(expectedPatient[key]);
    });
    expect(createdPatient._id).is.equal(id);
    expect(createdPatient._shardKey).is.equal(id);
```
from the create test, and use this to check the database properties. Also update the assertions to check the database properties are not returned.

File: src/PatientTestsApi/Test/Controllers/PatientController.spec.ts [hunk 2]
```diff
@@ -116,4 +124,46 @@ describe("PatientController", async function (): Promise<void> {
...
     expect(response.body).to.match(/^Error creating audit log:/i);
   });
+
+  it("Returns BadRequest if patient id not passed in.", async function (): Promise<void> {
+    const dataServiceMock = mock<IPatientDataService>();
+    const controller = createController(instance(dataServiceMock));
+    const request = createEmptyRequest();
+    
+    const response = await controller.findPatient(request);
+    expect(response).to.be.instanceOf(BadRequestResponse);
```
Comment (hannesne) [inline src/PatientTestsApi/Test/Controllers/PatientController.spec.ts:134]: Add an assertion to check the body has the expected message that id is missing.

**Comment by daniel-rocha** [comment]
@hannesne all comments resolved.

File: src/PatientTestsApi/LoadPatient/function.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "bindings": [
+    {
+      "authLevel": "anonymous",
```
Comment (linkinchow) [inline src/PatientTestsApi/LoadPatient/function.json:4]: @daniel-rocha @hannesne I think it's better to use "authLevel": "function" instead. That's the reason why APIM needs to retrieve the host key in key vault. We may need to change it in other function.json as well.

**Pull request merged**